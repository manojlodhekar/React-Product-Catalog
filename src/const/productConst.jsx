export default {
    PRODUCT_SEARCH_API_URL: 'http://api.walmartlabs.com/v1/search',
    PRODUCT_LOOKUP_API_URL: 'http://api.walmartlabs.com/v1/items',
    PRODUCT_RECCOMENDATION_API_URL: 'http://api.walmartlabs.com/v1/nbp?apiKey=',
    PRODUCT_API_KEY: 'afffhq7phcwhc28kb45skp9d',
    PRODUCT_MAIN_PAGE_SUBHEADER: 'Please search your products and click on it to view for more Reccomendations',
    PRODUCT_RECCOMENDATION_PAGE_SUBHEADER: 'More Reccomendated products to choose. Please click on Item to see more details',
    PRODUCT_DETAILS_PAGE_SUBHEADER: 'Details about',
    PLACEHOLDER_TEXT: 'Please enter product name ...',
    PRODUCT_IN_STOCK: 'Note : Product is available and in Stock right now',
    PRODUCT_OUT_STOCK: 'Note : Product is NOT available and out of Stock right now',
    SEARCHED_PRODUCT_MESSAGE: 'Searched products will load here...',
    HEADER_TEXT: 'Wallmart Products Warehouse',
    ERROR_MESSAGE: 'Oops..Something went Wrong, please try again.'
  
    }
    