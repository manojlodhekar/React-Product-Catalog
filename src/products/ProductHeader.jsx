import React from 'react';
import { Image, Grid, Divider } from 'semantic-ui-react'
import productConst from '../const/productConst';

const styles = {
  
    header:{
        paddingLeft: 10,
        textAlign: 'left',
        paddingBottom: 10,
        borderBottom: '1px solid grey'
    },
    headerTxt:{
        paddingTop: 10,
        textAlign: 'left',
        fontSize: 24,
        fontWeight: 'bold',
        lineHeight: 1.2
    },
    logoCls:{
        maxWidth: 60, 
        float: 'right', 
        textAlign: 'right'
    },
    subHeaderText:{
        color: '#0071ce',
        textAlign: 'left',
        fontSize: 18
    }

}

const ProductHeader = (props) =>{

    return(
        <div>
            <Grid >  
                <Grid.Row style={{paddingLeft: 20}}>
                    <Grid.Column width={3} style={{float: 'right'}}>
                        <Image src='/assets/Walmart_Spark.png' style={styles.logoCls} />
                    </Grid.Column>
                    <Grid.Column width={13}>
                        <div style={styles.headerTxt}>
                                {productConst.HEADER_TEXT}
                        </div>
                        <div style={styles.subHeaderText}>
                                {props.subHeaderText}
                        </div>
                    </Grid.Column>
                </Grid.Row>
            </Grid>
            <Divider />

           {/* <Header as='h2' style={styles.header}>
                    <Image src='/assets/Walmart_Spark.png' width="60"/> 
                    <Header.Content style={styles.headerTxt}>
                    {productConst.HEADER_TEXT}
                    <Header.Subheader>{props.subHeaderText}</Header.Subheader>
                    </Header.Content>
                </Header> */}
        </div>
    )
}
export default ProductHeader;