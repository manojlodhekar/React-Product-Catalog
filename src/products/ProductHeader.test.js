import React from 'react';
import ReactDOM from 'react-dom';
import ProductHeader from './ProductHeader';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import productConst from '../const/productConst';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });
let productHeader;
beforeEach(()=>{
  productHeader = shallow(<ProductHeader subHeaderText="This is subheader text" />);

})
describe('ProductHeader is defined', () => {

  it('should render correctly', () => {
    expect(productHeader).toHaveLength(1);
  });

  it('should render inner component', () => {
    expect(productHeader).toMatchSnapshot();
  });
  it('should contains passed message', () => {
    expect(productHeader.contains('This is subheader text')).toBe(true);
  });
  it('should contains default message', () => {
    expect(productHeader.contains(productConst.HEADER_TEXT)).toBe(true);
  });

 });
