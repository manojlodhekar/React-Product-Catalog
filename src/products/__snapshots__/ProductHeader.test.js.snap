// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductHeader is defined should render inner component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductHeader
    subHeaderText="This is subheader text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Grid>
          <GridRow
                    style={
                              Object {
                                        "paddingLeft": 20,
                                      }
                    }
          >
                    <GridColumn
                              style={
                                        Object {
                                                  "float": "right",
                                                }
                              }
                              width={3}
                    >
                              <Image
                                        as="img"
                                        src="/assets/Walmart_Spark.png"
                                        style={
                                                  Object {
                                                            "float": "right",
                                                            "maxWidth": 60,
                                                            "textAlign": "right",
                                                          }
                                        }
                                        ui={true}
                              />
                    </GridColumn>
                    <GridColumn
                              width={13}
                    >
                              <div
                                        style={
                                                  Object {
                                                            "fontSize": 24,
                                                            "fontWeight": "bold",
                                                            "lineHeight": 1.2,
                                                            "paddingTop": 10,
                                                            "textAlign": "left",
                                                          }
                                        }
                              >
                                        Wallmart Products Warehouse
                              </div>
                              <div
                                        style={
                                                  Object {
                                                            "color": "#0071ce",
                                                            "fontSize": 18,
                                                            "textAlign": "left",
                                                          }
                                        }
                              >
                                        This is subheader text
                              </div>
                    </GridColumn>
          </GridRow>
</Grid>,
        <Divider />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <GridRow
            style={
                        Object {
                                    "paddingLeft": 20,
                                  }
            }
>
            <GridColumn
                        style={
                                    Object {
                                                "float": "right",
                                              }
                        }
                        width={3}
            >
                        <Image
                                    as="img"
                                    src="/assets/Walmart_Spark.png"
                                    style={
                                                Object {
                                                            "float": "right",
                                                            "maxWidth": 60,
                                                            "textAlign": "right",
                                                          }
                                    }
                                    ui={true}
                        />
            </GridColumn>
            <GridColumn
                        width={13}
            >
                        <div
                                    style={
                                                Object {
                                                            "fontSize": 24,
                                                            "fontWeight": "bold",
                                                            "lineHeight": 1.2,
                                                            "paddingTop": 10,
                                                            "textAlign": "left",
                                                          }
                                    }
                        >
                                    Wallmart Products Warehouse
                        </div>
                        <div
                                    style={
                                                Object {
                                                            "color": "#0071ce",
                                                            "fontSize": 18,
                                                            "textAlign": "left",
                                                          }
                                    }
                        >
                                    This is subheader text
                        </div>
            </GridColumn>
</GridRow>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <GridColumn
                style={
                                Object {
                                                "float": "right",
                                              }
                }
                width={3}
>
                <Image
                                as="img"
                                src="/assets/Walmart_Spark.png"
                                style={
                                                Object {
                                                                "float": "right",
                                                                "maxWidth": 60,
                                                                "textAlign": "right",
                                                              }
                                }
                                ui={true}
                />
</GridColumn>,
              <GridColumn
                width={13}
>
                <div
                                style={
                                                Object {
                                                                "fontSize": 24,
                                                                "fontWeight": "bold",
                                                                "lineHeight": 1.2,
                                                                "paddingTop": 10,
                                                                "textAlign": "left",
                                                              }
                                }
                >
                                Wallmart Products Warehouse
                </div>
                <div
                                style={
                                                Object {
                                                                "color": "#0071ce",
                                                                "fontSize": 18,
                                                                "textAlign": "left",
                                                              }
                                }
                >
                                This is subheader text
                </div>
</GridColumn>,
            ],
            "style": Object {
              "paddingLeft": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Image
                  as="img"
                  src="/assets/Walmart_Spark.png"
                  style={
                                    Object {
                                                      "float": "right",
                                                      "maxWidth": 60,
                                                      "textAlign": "right",
                                                    }
                  }
                  ui={true}
/>,
                "style": Object {
                  "float": "right",
                },
                "width": 3,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "as": "img",
                  "src": "/assets/Walmart_Spark.png",
                  "style": Object {
                    "float": "right",
                    "maxWidth": 60,
                    "textAlign": "right",
                  },
                  "ui": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <div
                    style={
                                        Object {
                                                            "fontSize": 24,
                                                            "fontWeight": "bold",
                                                            "lineHeight": 1.2,
                                                            "paddingTop": 10,
                                                            "textAlign": "left",
                                                          }
                    }
>
                    Wallmart Products Warehouse
</div>,
                  <div
                    style={
                                        Object {
                                                            "color": "#0071ce",
                                                            "fontSize": 18,
                                                            "textAlign": "left",
                                                          }
                    }
>
                    This is subheader text
</div>,
                ],
                "width": 13,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Wallmart Products Warehouse",
                    "style": Object {
                      "fontSize": 24,
                      "fontWeight": "bold",
                      "lineHeight": 1.2,
                      "paddingTop": 10,
                      "textAlign": "left",
                    },
                  },
                  "ref": null,
                  "rendered": "Wallmart Products Warehouse",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "This is subheader text",
                    "style": Object {
                      "color": "#0071ce",
                      "fontSize": 18,
                      "textAlign": "left",
                    },
                  },
                  "ref": null,
                  "rendered": "This is subheader text",
                  "type": "div",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Grid>
            <GridRow
                        style={
                                    Object {
                                                "paddingLeft": 20,
                                              }
                        }
            >
                        <GridColumn
                                    style={
                                                Object {
                                                            "float": "right",
                                                          }
                                    }
                                    width={3}
                        >
                                    <Image
                                                as="img"
                                                src="/assets/Walmart_Spark.png"
                                                style={
                                                            Object {
                                                                        "float": "right",
                                                                        "maxWidth": 60,
                                                                        "textAlign": "right",
                                                                      }
                                                }
                                                ui={true}
                                    />
                        </GridColumn>
                        <GridColumn
                                    width={13}
                        >
                                    <div
                                                style={
                                                            Object {
                                                                        "fontSize": 24,
                                                                        "fontWeight": "bold",
                                                                        "lineHeight": 1.2,
                                                                        "paddingTop": 10,
                                                                        "textAlign": "left",
                                                                      }
                                                }
                                    >
                                                Wallmart Products Warehouse
                                    </div>
                                    <div
                                                style={
                                                            Object {
                                                                        "color": "#0071ce",
                                                                        "fontSize": 18,
                                                                        "textAlign": "left",
                                                                      }
                                                }
                                    >
                                                This is subheader text
                                    </div>
                        </GridColumn>
            </GridRow>
</Grid>,
          <Divider />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <GridRow
              style={
                            Object {
                                          "paddingLeft": 20,
                                        }
              }
>
              <GridColumn
                            style={
                                          Object {
                                                        "float": "right",
                                                      }
                            }
                            width={3}
              >
                            <Image
                                          as="img"
                                          src="/assets/Walmart_Spark.png"
                                          style={
                                                        Object {
                                                                      "float": "right",
                                                                      "maxWidth": 60,
                                                                      "textAlign": "right",
                                                                    }
                                          }
                                          ui={true}
                            />
              </GridColumn>
              <GridColumn
                            width={13}
              >
                            <div
                                          style={
                                                        Object {
                                                                      "fontSize": 24,
                                                                      "fontWeight": "bold",
                                                                      "lineHeight": 1.2,
                                                                      "paddingTop": 10,
                                                                      "textAlign": "left",
                                                                    }
                                          }
                            >
                                          Wallmart Products Warehouse
                            </div>
                            <div
                                          style={
                                                        Object {
                                                                      "color": "#0071ce",
                                                                      "fontSize": 18,
                                                                      "textAlign": "left",
                                                                    }
                                          }
                            >
                                          This is subheader text
                            </div>
              </GridColumn>
</GridRow>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <GridColumn
                  style={
                                    Object {
                                                      "float": "right",
                                                    }
                  }
                  width={3}
>
                  <Image
                                    as="img"
                                    src="/assets/Walmart_Spark.png"
                                    style={
                                                      Object {
                                                                        "float": "right",
                                                                        "maxWidth": 60,
                                                                        "textAlign": "right",
                                                                      }
                                    }
                                    ui={true}
                  />
</GridColumn>,
                <GridColumn
                  width={13}
>
                  <div
                                    style={
                                                      Object {
                                                                        "fontSize": 24,
                                                                        "fontWeight": "bold",
                                                                        "lineHeight": 1.2,
                                                                        "paddingTop": 10,
                                                                        "textAlign": "left",
                                                                      }
                                    }
                  >
                                    Wallmart Products Warehouse
                  </div>
                  <div
                                    style={
                                                      Object {
                                                                        "color": "#0071ce",
                                                                        "fontSize": 18,
                                                                        "textAlign": "left",
                                                                      }
                                    }
                  >
                                    This is subheader text
                  </div>
</GridColumn>,
              ],
              "style": Object {
                "paddingLeft": 20,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Image
                    as="img"
                    src="/assets/Walmart_Spark.png"
                    style={
                                        Object {
                                                            "float": "right",
                                                            "maxWidth": 60,
                                                            "textAlign": "right",
                                                          }
                    }
                    ui={true}
/>,
                  "style": Object {
                    "float": "right",
                  },
                  "width": 3,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "as": "img",
                    "src": "/assets/Walmart_Spark.png",
                    "style": Object {
                      "float": "right",
                      "maxWidth": 60,
                      "textAlign": "right",
                    },
                    "ui": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <div
                      style={
                                            Object {
                                                                  "fontSize": 24,
                                                                  "fontWeight": "bold",
                                                                  "lineHeight": 1.2,
                                                                  "paddingTop": 10,
                                                                  "textAlign": "left",
                                                                }
                      }
>
                      Wallmart Products Warehouse
</div>,
                    <div
                      style={
                                            Object {
                                                                  "color": "#0071ce",
                                                                  "fontSize": 18,
                                                                  "textAlign": "left",
                                                                }
                      }
>
                      This is subheader text
</div>,
                  ],
                  "width": 13,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Wallmart Products Warehouse",
                      "style": Object {
                        "fontSize": 24,
                        "fontWeight": "bold",
                        "lineHeight": 1.2,
                        "paddingTop": 10,
                        "textAlign": "left",
                      },
                    },
                    "ref": null,
                    "rendered": "Wallmart Products Warehouse",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "This is subheader text",
                      "style": Object {
                        "color": "#0071ce",
                        "fontSize": 18,
                        "textAlign": "left",
                      },
                    },
                    "ref": null,
                    "rendered": "This is subheader text",
                    "type": "div",
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
